CodeEdit/Utils/Extensions/String/String+HighlightOccurrences.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  String+HighlightOccurrences.swift
+//  CodeEdit
+//
+//  Created by Raymond Vleeshouwer on 13/06/23.
+//
+
+import Foundation
+import SwiftUI
+
+extension String {
+    /// Highlights occurences of a substring in a string and returns text highlighted as such
+    func highlightOccurrences(_ ofSearch: String) -> some View {
+        if ofSearch.isEmpty {
+            return Text(self)
+        }
+
+        let ranges = self.rangesOfSubstring(ofSearch.lowercased())
+
+        var currentIndex = self.startIndex
+        var highlightedText = Text("")
+
+        for range in ranges {
+            let nonHighlightedText = self[currentIndex..<range.lowerBound]
+            let highlightedSubstring = self[range]
+
+            // swiftlint:disable shorthand_operator
+            highlightedText = highlightedText + Text(nonHighlightedText).foregroundColor(.secondary)
+            highlightedText = highlightedText + Text(highlightedSubstring).foregroundColor(.primary)
```

Comment:
I would suggest to make it bold.
on my machine the text color change make no clear visual difference on screen.

```suggestion
            highlightedText = highlightedText + Text(highlightedSubstring).foregroundColor(.primary).bold()
```

<table><tr><th>Without Bold</th><th>With <strong>bold()</strong></th></tr>
<td>
<img width="220" alt="image" src="https://github.com/CodeEditApp/CodeEdit/assets/1290461/df50ff8d-f29f-4286-a655-56c1ddfa37c3">

</td><td>
<img width="209" alt="image" src="https://github.com/CodeEditApp/CodeEdit/assets/1290461/ae5bef4f-94a7-4152-bae9-1f551f7d2b24">

</td></tr></table>